File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        
        List<String> str = new ArrayList<>();
        
        for (int i = 0; i < numbers.length; i++) {
            str.add("" + numbers[i]);
        }
        
        str.sort((s1, s2) -> (s2 + s1).compareTo(s1 + s2));
        
        if (str.get(0).equals("0")) return "0";
        
        StringBuilder sb = new StringBuilder();
        for (String item : str) sb.append(item);
        
        return sb.toString();
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <string>
#include <algorithm>
#include <iostream>

using namespace std;

string solution(vector<int> numbers) {
    string answer = "";
    sort(numbers.begin(), numbers.end(), [](
    	const int& a, const int& b) {
        string sa = to_string(a), sb = to_string(b);
        return sa + sb > sb + sa;
    });
    for (auto& item : numbers) answer += to_string(item);
    
    // prefix로 붙은 0은 제거
    if (answer[0] == '0') {
        int i = 0;
        while (true) {
            if (answer[i] == '0') i++;
            else break;
        }
        answer.erase(0, i);
        if (answer.size() == 0) answer = "0";
    }
    
    return answer;
}